VTUL/dlp-access
# Handles asset relative-urls with no domain to make site backend env agnostic

**Handles asset relative-urls with no domain to make site backend env agnostic**
* * *

**JIRA Ticket**: ([link](https://webapps.es.vt.edu/jira/browse/LIBTD-2643)) (:star:)

* Other Relevant Links (Meeting note, project page, related pull requests, etc.)

# What does this Pull Request do? (:star:)
Handles asset relative-urls with no domain so that DB records don't point to old (and possibly deleted) s3 buckets that the app does not have access to. This PR also updates a boatload of tests and refactors code to support more reliable testing. Finally, this PR refactors the admin section to notify the application when a site record gets modified so that the application can fetch the newest version

# What's the changes? (:star:)
* Adds domain to relative urls in the database so that application will request them from the correct bucket
* Refactors to save urls to DB without domain attached
* Refactor tests to be more reliable
* Refactor admin section to notify application of changes to site record so that application can fetch changes.

# How should this be tested?
* Check that assets continue to load correctly
* Check that multi-media section on homepage updates immediately after changes to the "Homepage Media Section" form without a browser refresh.

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

* branch: `LIBTD-2643`

# Interested parties
@andreaWaldren 

(:star:) Required fields
